pandas/core/frame.py

```diff
@@ -4382,17 +4382,17 @@ def round(self, decimals=0, out=None):
         from pandas.tools.merge import concat
 
         def _dict_round(df, decimals):
-            for col in df:
+            for col, vals in df.iteritems():
                 try:
-                    yield np.round(df[col], decimals[col])
+                    yield np.round(vals, decimals[col])
```

Comment:
The value of decimals is either of type dict or pandas.Series. Whereas the former can not have duplicate entries, a pandas.Series might have and direct indexing will return all values with the same index. So, this case is still not handled.